### PR TITLE
Dev container

### PR DESCRIPTION
**Pull request by nojaf**
As a response to @gubser poor experience in https://github.com/fsprojects/fantomas/issues/2972, I tried to set up a devcontainer.

While doing this, I did encounter some problems when only `8rc2` SDK was installed.
I needed `RollForward` for the test projects and of course, had to workaround https://github.com/dotnet/sdk/issues/35989.

During my experiment, I updated `fsdocs` to the latest alpha. Even though we use a custom theme, I had to change some things for the search.
<strike>
Ionide was not starting in my dev container. @Martin521 would you have any ideas what I'm still missing?
</strike>

### CODE REVIEW

**Comment by nojaf** [comment]
image

**Comment by nojaf** [comment]
Turns out, that I was trying to use a local FSAC inside the container.
Works now.

**Comment by dawedawe** [comment]
Mmh, so far my success rate with the container is 2 out of 10 but if it can unblock others, sure, let's have it.

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	// "features": {},
+
+	// Configure tool-specific properties.
+	"customizations": {
+		// Configure properties specific to VS Code.
+		"vscode": {		
+			// Add the IDs of extensions you want installed when the container is created.
+			"extensions": [
+				"Ionide.Ionide-fsharp",
+				"ms-dotnettools.csharp",
```
Comment (dawedawe) [inline .devcontainer/devcontainer.json:18]: Not sure if it helps, but maybe have the csharp extension before Ionide, as Ionide depends on it.

**Comment by Martin521** [comment]
This is how dotnet/fsharp does it:

```
    "customizations": {
        "vscode": {
            // Add the IDs of extensions you want installed when the container is created.
            "extensions": [
                "ms-dotnettools.csharp",
                "Ionide.Ionide-fsharp",
                "tintoy.msbuild-project-tools"
            ]
        }
    },
```

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (nojaf) [reply .devcontainer/devcontainer.json:18]: Yeah, the order doesn't seem to matter.

**Comment by nojaf** [comment]
Yup, stole that part from `dotnet/fsharp` indeed.

**Comment by Martin521** [comment]
And wouldn't it make sense to use a .net 8 image?

**Comment by nojaf** [comment]
> And wouldn't it make sense to use a .net 8 image?

Yes, probably, I selected the `F#` option from the UI.
Might make sense to use a Dockerfile instead.

**Comment by nojaf** [comment]
You mean in Windows or in the container?

**Comment by dawedawe** [comment]
In the container

**Comment by nojaf** [comment]
Ok, I now have it in Windows. This is a checkout-as line ending shenannigan.
Forcing `lf` in the test now.

**Review by dawedawe — APPROVED** [review]
Much more stable for me now 👍

**Pull request merged**